lte/gateway/c/core/oai/include/map.h

```diff
@@ -0,0 +1,228 @@
+      return "MAP_EMPTY";
+      break;
+
+    case MAP_DUMP_FAIL:
+      return "MAP_DUMP_FAIL";
+      break;
+
+    default:
+      return "UNKNOWN map_rc_t";
+  }
+}
+
+/***************************************************************************
+**                                                                        **
+** Name:    map_ts_s                                                      **
+**                                                                        **
+** Description: This is a generic structure for maps.It is implemented    **
+**              using template struct definitions.                        **
+**                                                                        **
+** Parameters:  keyT     - data type of key                               **
+**              valueT   - data type of value                             **
+**              Hash     - used to pass a custom hash function.           **
+**                         Default: std::hash<keyT>                       **
+**              KeyEqual - it is used to overload the EqualTo operator.   **
+**                         When using a userdefined struct as key,        **
+**                         pass a suitable EqualTo method.                **
+**                            Default: std::equal_to<keyT>                **
+**                                                                        **
+** APIs:       set_name() , get_name(), get(), insert(), delete()        **
+**                                                                        **
+***************************************************************************/
+template<
+    typename keyT, typename valueT, class Hash = std::hash<keyT>,
+    class KeyEqual = std::equal_to<keyT>>
+struct map_ts_s {
+  std::mutex mutex_obj;
+  std::unordered_map<keyT, valueT, Hash, KeyEqual> umap;
+  std::string name;
+  bool log_enabled;
+
+  void set_name(std::string umap_name) { name = umap_name; }
+  std::string get_name() { return name; }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    get()                                                         **
+  **                                                                        **
+  ** Description: Takes key and valueP as parameters.If the key exists,     **
+  **              corresponding value is returned through the valueP,       **
+  **              else returns error.                                       **
+  **                                                                        **
+  ***************************************************************************/
+  map_rc_t get(const keyT key, valueT* valueP) {
+    if (umap.empty()) {
+      return MAP_EMPTY;
+    }
+    if (valueP == nullptr) {
+      return MAP_BAD_PARAMETER_VALUE;
+    }
+    std::lock_guard<std::mutex> lock(mutex_obj);
```

Comment:
@ardzoht do we need thread safety as each hash map is/should be local to the thread?

Reply:
That's a good callout @ulaskozat, I think these can be removed following the expectation that the hashtables should be only used by its owner state thread. Given that we will iteratively will migrate these hashtables, we can follow that expectation during the process of the migration. @shashidhar-patil Let's remove the locking and update the name of the struct accordingly.

Reply:
That's a good callout @ulaskozat, I think these can be removed following the expectation that the hashtables should be only used by its owner state thread. Given that we will iteratively will migrate these hashtables, we can follow that expectation during the process of the migration. @shashidhar-patil Let's remove the locking and update the name of the struct accordingly.

Reply:
@ardzoht Okay will do that.

Reply:
Addressed this comment in the newest commit.